python_modules/dagster/dagster/_core/definitions/assets_job.py

```diff
@@ -223,35 +225,159 @@ def asset2(asset1):
     )
 
 
-def build_job_partitions_from_assets(
-    assets: Iterable[Union[AssetsDefinition, SourceAsset]],
-) -> Optional[PartitionsDefinition]:
-    assets_with_partitions_defs = [assets_def for assets_def in assets if assets_def.partitions_def]
+def get_asset_graph_for_job(
+    parent_asset_graph: AssetGraph, selection: AssetSelection
+) -> AssetGraph:
+    """Subset an AssetGraph to create an AssetGraph representing an asset job.
```

Comment:
I think this could be named better. `get_asset_graph_for_job` is overly specific. Seems like the name should be `create_asset_graph_of_selection` or something.